plugins/modules/packaging/language/npm.py

```diff
@@ -193,14 +193,18 @@ def _exec(self, args, run_in_check_mode=False, check_rc=True):
 
             rc, out, err = self.module.run_command(cmd, check_rc=check_rc, cwd=cwd)
             return out
-        return ''
+        return '{}'
```

Comment:
Is this change necessary?

Reply:
yes

```
>>> a = ''
>>> import json
>>> json.loads(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
>>> json.loads('{}')
{}
```

Reply:
There's only one piece of code that uses `json.loads` of the result, and that code passes `run_in_check_mode=True`. When `run_in_check_mode` is `True`, this line is never hit.

Reply:
(Also a way better workaround would be `data = json.loads(self._exec(cmd, True, False) or '{}')` instead of modifying the general return value of `self._exec`.)